File: samples/Python/zixipay-merchant-api.py
import requests
import urllib.parse
import hmac
import hashlib
import time
import json

#
# Zixipay API endpoint call
#
# endpoint: string, endpoint you want to call
# params: array, enpoint parameters
# uid: string, User ID
# apikey: string, API Access Key
#

def ZixiPay_ApiCall(endpoint, params, apikey):
    apiurl = 'https://api.zixipay.com/apiv2/'    # API endpoint URL
    payload = urllib.parse.urlencode(params)    # create urlencoded payload
    signature = hmac.new(bytes(apikey, 'latin-1'), bytes(payload, 'latin-1'), hashlib.sha256).hexdigest().lower()    # generate signature
    params.append(('sig', signature))    # add signature to the end of api parameters array
    result = requests.post(apiurl+endpoint, data = params, headers = {"Content-Type": "application/x-www-form-urlencoded"})
    if result == None:
        return False
    else:
        return json.loads(result.text)


uid = ''          # ZixiPay User ID
apikey = ''       # ZixiPay API Access Key

# API call parameters 
prms = [
    ('currency',  'BTC'),
    ('ref',  'anyreference'),
    ('uid',  uid),
    ('ts',   int(time.time()))
]

# getpaymentwallet call
result = ZixiPay_ApiCall('getpaymentwallet', prms, apikey)
print(result)

File: samples/JS/zixipay-merchant-api.js
var axios = require('axios');
var crypto = require('crypto');
var querystring = require('querystring');

var uid = '';          // ZixiPay User ID
var api_key = '';       // ZixiPay API Access Key

prms = {
	uid: uid,
	ts: Math.floor(Date.now() / 1000)
};

// getpaymentwallet call
ZixiPay_ApiCall('getpaymentwallet', prms, api_key).then((r) => {
	if(r.result == 'ok')
		console.log(r.payload);
});


/**
 * Zixipay API endpoint call
 */
async function ZixiPay_ApiCall(endpoint, params, apikey) {
	
	api_url = 'https://api.zixipay.com/apiv2/';   // ZixiPay API endpoint URL

	params.sig = crypto.createHmac("sha256", apikey).update(querystring.stringify(params)).digest("hex");

	return await axios.post(api_url+endpoint, querystring.stringify(params), { headers: { "Content-Type": "application/x-www-form-urlencoded" } })
		.then((r) => {
			return r.data;
		})
		.catch((r) => {
			//handle error
			return false;
	});
}
